app/services/interface_cleaner.rb

```diff
@@ -0,0 +1,51 @@
+class InterfaceCleaner
+  include ActiveRecord::Sanitization
+
+  ESCAPE_CHAR = '\\'
+
+  def self.nic_pattern_to_like(nic)
+    sanitize_sql_like(nic, ESCAPE_CHAR).tr('*', '%')
+  end
+
+  def self.ignored_interface_like_patterns
+    Setting[:ignored_interface_identifiers].map { |pattern| nic_pattern_to_like(pattern) }
+  end
+
+  delegate :logger, :to => :Rails
+
+  attr_reader :primary_nics, :provision_nics, :primary_hosts, :provision_hosts
+
+  def clean!
+    @deleted_count = delete_excluded_nics
+
+    @primary_nics, @primary_hosts = excluded_nics.where(primary: true).pluck(:id, :host_id).transpose
+    @provision_nics, @provision_hosts = excluded_nics.where(provision: true).pluck(:id, :host_id).transpose
```

Comment:
Can you make this a memorized class variable via reader methods rather than creating this every time `clean!` is performed? Or even just attribute method without memorizing, if that's really only for testing.

Reply:
This is done after the main deletion so I don't expect too much records here, but if you insist I can make those properties lazy loaded.

Reply:
That's fine.